Documentation/proposals/202309-controller-id.md

```diff
@@ -0,0 +1,65 @@
+# Controller-ID Proposal
+
+## Owners:
+* [danielmellado](https://github.com/danielmellado)
+
+## Related Tickets:
+* [#4281](https://github.com/prometheus-operator/prometheus-operator/issues/4281)
+* [#4498](https://github.com/prometheus-operator/prometheus-operator/pull/4498)
+
+## Other Docs:
+* n/a
+
+### Introduction
+
+This proposal aims to implement a solution to support multiple cluster-level
+Prometheus instances running concurrently without conflicting over the same CRD.
+This issue can significantly impact use cases where multiple Prometheus operator
+instances run in parallel.
+
+## Why
+
+Currently, we encounter issues when users or different operators deploy Prometheus
+alongside Prometheus Operator's instance. Even though User-Managed Workloads
+(UWM) exists, some operators or users may deploy Prometheus directly from
+manifests, Helm charts, or other rogue operators, leading to the creation of
+'rogue' Prometheus instances alongside the operator's instance.
+
+In the worst-case scenario, these operators may not only compete for ownership
+of the CRD resources but also attempt to rewrite or redeploy different versions
+of the CRD, causing disruptions to all pods.
+
+The remediation for this scenario, where users deploy their Prometheus operator
+instances in parallel, involves using one of the many CLI arguments, such as
+`--deny-namespaces`, `--namespaces`, `--prometheus-instance-selector`, or
+`prometheus-instance-namespaces`.
+
+Although the initial discussion for this proposal considered adding an owner
+reference within the scope, the
+[ControllerRef](https://github.com/kubernetes/design-proposals-archive/blob/acc25e14ca83dfda4f66d8cb1f1b491f26e78ffe/api-machinery/controller-ref.md)
+model does not directly address this problem.
+
+## How
+
+After some research, we have identified a potential solution already
+implemented by the
+[zalando/postgres-operator](https://github.com/zalando/postgres-operator). When
+an operator is configured with a specific "controller ID" value, it will only
+reconcile resources that have a matching "controller ID" annotation.
+
+Conversely, if the operator is not configured with a "controller ID," it will
+skip all resources that have a "controller ID" annotation. More details can be
+found in the
+[zalando/postgres-operator documentation](https://github.com/zalando/postgres-operator/blob/master/docs/administrator.md#operators-with-defined-ownership-of-certain-postgres-clusters).
+
+## Goals
+
+* Add a similar "controller-id" based approach to PO.
```

Comment:
I'd rather phrase it as: "Guarantee that a custom resource will be managed by a specific Prometheus operator instance"